repairnator/src/main/java/fr/inria/spirals/repairnator/helpers/GitHelper.java

```diff
@@ -0,0 +1,190 @@
+import org.eclipse.jgit.errors.IncorrectObjectTypeException;
+import org.eclipse.jgit.errors.MissingObjectException;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.kohsuke.github.*;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.List;
+
+/**
+ * Created by fernanda on 01/03/17.
+ */
+public class GitHelper {
+
+    /**
+     * Test if a commit exists in the given git repository
+     *
+     * @param git
+     * @param oldCommitSha
+     * @return oldCommitSha if the commit exists in the repo, a new commit SHA
+     *         if the commit has been retrieved from GitHub and applied back, or
+     *         null if the retrieve failed.
+     */
+    public static String testCommitExistence(Git git, String oldCommitSha, AbstractStep step, Build build) {
+        try {
+            ObjectId commitObject = git.getRepository().resolve(oldCommitSha);
+            git.getRepository().open(commitObject);
+            return oldCommitSha;
+        } catch (MissingObjectException e) {
+            return retrieveAndApplyCommitFromGithub(git, oldCommitSha, step, build);
+        } catch (IncorrectObjectTypeException e) {
+            step.addStepError("Error while testing commit: " + e);
+        } catch (AmbiguousObjectException e) {
+            step.addStepError("Error while testing commit: " + e);
+        } catch (IOException e) {
+            step.addStepError("Error while testing commit: " + e);
+        }
+        return null;
+    }
+
+    /**
+     * When a commit has been force deleted it still can be retrieved from
+     * GitHub API. This function intend to retrieve a patch from the Github API
+     * and to apply it back on the repo
+     *
+     * @param git
+     * @param oldCommitSha
+     * @return the SHA of the commit created after applying the patch or null if
+     *         an error occured.
+     */
+    private static String retrieveAndApplyCommitFromGithub(Git git, String oldCommitSha, AbstractStep step, Build build) {
+        try {
+            GitHub gh = GitHubBuilder.fromEnvironment().build();
+            GHRepository ghRepo = gh.getRepository(build.getRepository().getSlug());
+
+            String lastKnowParent = getLastKnowParent(gh, ghRepo, git, oldCommitSha, step);
+
+            // checkout the repo to the last known parent of the deleted commit
+            git.checkout().setName(lastKnowParent).call();
+
+            // get from github a patch between that commit and the targeted
+            // commit
+            // note that this patch could contain changes of multiple commits
+            GHCompare compare = ghRepo.getCompare(lastKnowParent, oldCommitSha);
+
+            showGitHubRateInformation(gh, step);
+
+            URL patchUrl = compare.getPatchUrl();
+
+            step.getLogger().debug("Retrieve commit patch from the following URL: " + patchUrl);
```

Comment:
You should use a logger of the current class, and display in the message the step name.